scripts: include newline before appending to bashrc

ibreakthecloud opened the pull request:
`pnpm` tool is installed in gceworker as a part of bootstraping the gceworker vm. This installation
also writes to `~/.bashrc` file inside the same vm without putting the trailing newline. This resulted
in new appends happening at the wrong line, which effectively rendered any new command append a code comment.
example:
before: `command1` is a new append

`# pnpm ends herecommand1`

after: `command1` is a new append
`#pnpm ends here`
`command1`

This commit adds newline at two different instances.
First after the installation of `pnpm` and before
appending `. ~/.bashrc_bootstrap`.

Part of DEVINF-1371

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139309)
<!-- Reviewable:end -->

jlinder commented on build/bootstrap/bootstrap-debian.sh:
```diff
@@ -32,6 +32,9 @@ sudo apt-get install -y --no-install-recommends \
 curl -fsSL https://get.pnpm.io/install.sh | env PNPM_VERSION=8.6.6 sh -
 
+# append new line to bashrc
+echo -e '\n' >> ~/.bashrc
```
How about either moving this to before line 42 (`echo '. ~/.bashrc_bootstrap' >> ~/.bashrc`) where it's necessity is clear on its own or stating in the comment why the new line is necessary? (If I were doing it, I'd move it to be before line 42.)

jlinder replied:
I see. That is true. But I'd also expect they would be testing it to make sure it works as expected, so they'd catch it. 

Another thing that would make it easier to understand the generated file would be to have a blank line between the pnpm installed code and the `. ~/.bashrc_bootstrap` line. That makes it easier to see the blocks of functionality and has a visual separation of the sourcing of .bashrc_bootstrap from the pnpm code. So putting a blank line in both places would be a good idea. What do you think of this idea?

ibreakthecloud replied:
I like this idea! I will fix this.

jlinder reviewed (changes requested):
A suggestion and a few notes:

- In the commit message text:
    - the `, when bootstrap-debian.sh,` part of the description would benefit from more context about `when`. A few possible examples: `, when creating bootstrap-debian.sh,` or `, during bootstrap-debian.sh creation,`.
    - the first sentence is really long and can be split in two to make it easier to read. For example: `at the end. This resulted in the append happening at the wrong line` 
    - The code quoted in the commit message doesn't show the problem. I'd suggest either showing the problem and state it's the code that shows the problem, or remove the code.
- The `Epic: none` reference is only necessary when there is no issue reference. See [this note](https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/2009039063/Issue+and+Epic+Refs+in+Commits+and+PRs).
- In the commit message, the value to use for the issue identifier should be just the issue ID (DEVINF-1371 in this case) or a URL to the issue. [Here](https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/2009039063/Issue+and+Epic+Refs+in+Commits+and+PRs) is the format reference.

ibreakthecloud committed:
scripts: include newline before appending to bashrc

`pnpm` tool is installed in gceworker as a part of
bootstraping the gceworker vm. This installation
also writes to `~/.bashrc` file inside the same vm
without putting the trailing newline. This resulted
in new appends happening at the wrong line, which
effectively rendered any new command append a code
comment.
example:
before: `command1` is a new append

`# pnpm ends herecommand1`

after: `command1` is a new append
`#pnpm ends here`
`command1`

This commit adds newline at two different instances.
First after the installation of `pnpm` and before
appending `. ~/.bashrc_bootstrap`.

Part of DEVINF-1371

Release note: none

ibreakthecloud commented:
@jlinder I have done few changes, PTAL.

ibreakthecloud commented:
TFTR @jlinder 

bors r+

ibreakthecloud commented:
bors r+
